utils/version_test.go

```diff
@@ -0,0 +1,71 @@
+package utils
+
+import "testing"
+
+func TestParseVersion(t *testing.T) {
+	// > 3 parts of version
+	if _, err := ParseVersion("1.1.1.1"); err != ErrInvalidVersionFormat {
+		t.Fatalf("Expected error: %s, took: %s", ErrInvalidVersionFormat.Error(), err)
+	}
+
+	//  < 3 parts of version
+	if _, err := ParseVersion("1.1"); err != ErrInvalidVersionFormat {
+		t.Fatalf("Expected error: %s, took: %s", ErrInvalidVersionFormat.Error(), err)
+	}
+
+	//  empty version part
+	if _, err := ParseVersion("1.1."); err == nil {
+		t.Fatal("Expected error but took nil")
+	}
+}
+
+func getVersion(v string, t *testing.T) *Version {
+	version, err := ParseVersion(v)
+	if err != nil {
+		t.Fatal(err)
+	}
+	return version
+}
+
+func TestVersion_Compare(t *testing.T) {
+	testData := []struct {
+		v1, v2 string
+		result ComparisonStatus
+	}{
+		{"0.0.0", "0.0.0", Equal},
+		{"1.0.0", "0.0.0", Greater},
+		{"0.0.0", "1.0.0", Less},
+
+		{"0.1.0", "0.0.0", Greater},
```

Comment:
```suggestion
		{"0.1.0", "0.0.0", Greater},
		{"0.84.2", "0.1.0", Greater},
		{"0.84.2", "1.0.0", Less},
```

Reply:
added (but imho it test the same cases : )